app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentPresenter.kt

```diff
@@ -120,19 +123,17 @@ class LocalPaymentPresenter(private val view: LocalPaymentView,
   }
 
   private fun handleOkErrorButtonClick() {
-    disposables.add(
-        view.getOkErrorClick()
-            .observeOn(viewScheduler)
-            .doOnNext { view.dismissError() }
-            .subscribe())
+    disposables.add(view.getErrorDismissClick()
+        .observeOn(viewScheduler)
+        .doOnNext { view.dismissError() }
+        .subscribe())
   }
 
   private fun handleOkBuyButtonClick() {
-    disposables.add(
-        view.getGotItClick()
-            .observeOn(viewScheduler)
-            .doOnNext { view.close() }
-            .subscribe()
+    disposables.add(view.getGotItClick()
+        .observeOn(viewScheduler)
+        .doOnNext { view.close() }
```

Comment:
i think this observe on is unnecessary since the method is always called in the main thread